File: Programmers/자바기초/전국대회선발고사.java
import java.util.*;

class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0 ; i < rank.length ; i++) {
            if (attendance[i]) {
                map.put(rank[i], i);
            }
        }
        
        ArrayList<Integer> keySet = new ArrayList<>(map.keySet());
        Collections.sort(keySet);
        
        int answer = 0;
        int idx = 0;
        for (Integer key : keySet) {
            if (idx == 0) {
                answer += (map.get(key) * 10000);
            } else if (idx == 1) {
                answer += (map.get(key) * 100);
            } else if (idx == 2) {
                answer += map.get(key);
            } else {
                break;
            }
            
            idx++;
        }
        
        
        
        return answer;
    }
}

File: Programmers/제로베이스_문제추천_py/레벨0/전국대회선발고사.py
def solution(rank, attendance):
    answer = 0
    students = []
    for i in range(len(rank)) :
        if attendance[i] :
            students.append((rank[i],i))
    
    students.sort(key = lambda x : x[0])
    answer = 10000 * students[0][1] + 100 * students[1][1] + students[2][1]
    return answer